Steps4Impact/NotificationsV2/NotificationsViewController.swift

```diff
@@ -31,35 +31,96 @@ import UIKit
 import RxSwift
 
 class NotificationsViewController: TableViewController {
+  
   override func commonInit() {
     super.commonInit()
 
     title = Strings.Notifications.title
     dataSource = NotificationsDataSource()
   }
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    checkNotificationPermission()
+  }
+  
+  override func reload() {
+    dataSource?.reload { [weak self] in
+      self?.tableView.reloadOnMain()
+    }
+  }
+  
+  override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+    super.tableView(tableView, willDisplay: cell, forRowAt: indexPath)
+    if let cell = cell as? NotificationPermissionCell {
+      cell.delegate = self
+    }
+  }
+  
+  func checkNotificationPermission() {
+    func reloadIfNeeded() {
```

Comment:
this whole method should be in the datasource and the view controller should call it